ArraysAndLists/Program.cs

```diff
@@ -18,46 +18,104 @@ public static class ArraysAndLists
         {
-            // Delete the below line when you are writing your code
-            return default;
+            if (numbers.Length != numbers.Distinct().Count())
+            {
+                return "Error";
+            }
+            Array.Sort(numbers);
+
+            var result = string.Empty;
+
+            foreach (var number in numbers)
+            {
+                result += number + " ";
+            }
+            return result.Trim();
+
         }
 
         // 3 - Write a method that accepts a string for a name. Use an array to reverse the characters
         // then return the result as a string. "I like trains" should return "sniart ekil I". Return
         // "Error" if no text is inputted.
         public static string ReverseName(string name)
         {
-            // Delete the below line when you are writing your code
-            return default;
+            char[] backwardArray = name.ToCharArray();
+
+            if (backwardArray.Length < 1)
+            {
+                return "Error";
+            }
+            else
+            {
+                Array.Reverse(backwardArray);
+            }
+            return new string(backwardArray);
         }
 
         // 4 - Write a method that accepts any number of integers. If none or less than 5 numbers,
         // return "Invalid Array"; otherwise, return the 3 smallest numbers as a string seperated by
         // spaces, sorted in ascending order. Treat duplicate numbers normally.
         public static string SmallestNumbers(params int[] numbers)
         {
-            // Delete the below line when you are writing your code
-            return default;
+            if (numbers.Length < 5)
+            {
+                return "Invalid Array";
+            }
+            else
+            {
+                Array.Sort(numbers);
+                return numbers[0] + " " + numbers[1] + " " + numbers[2];
+            }
         }
 
         // 5 - Write a method that accepts any number of numbers (assume at least two numbers). The numbers
         // may include duplicates. Return the distinct numbers as an array sorted by ascending order.
         // So for example, an array consisting of [2, 2, 1, 3] should return [1, 2, 3].
         public static int[] UniqueNumbers(params int[] numbers)
         {
-            // Delete the below line when you are writing your code
-            return default;
-        }
+            /* Array.Sort(numbers);
 
+            return numbers.Distinct().ToArray();*/
+            
+            var numbersList = numbers.ToList();
+
+            for (int i = 0; i < numbersList.Count; i++)
+            {
+                for (var j = i ; j < numbersList.Count; j++)
+                {
+                    if (numbersList[i] == numbersList[j])
+                    {
+                        numbersList.RemoveAt(j);
+                    }
+                }
+            }
```

Comment:
Just tested this, it doesn't run properly when you use 

` var result = ArraysAndLists.ArraysAndLists.UniqueNumbers(1, 2, 5, 5, 3, 3, 3, 7, 8);`

As the input (you can try it out yourself. The results don't match the tests. You helped to flag an oversight! :) We basically forgot to check something.

The correct solution is to change the second 4 loop to,

``for (var j = i + 1 ; j < numbersList.Count; j++)`` @ line 107.

and to stick a ``j--;`` after line 111.